### ISSUE TITLE
Missing `securityScheme` type in OAuth2 documentation

### ISSUE THREAD

**Issue by VladDrakul1986** [javalin/javalin-openapi#56]
thx for your efforts to rework the old plugin.

I figured out one bug when using your example java code.

The generated json is missing one important field when using OAuth2 as security scheme.
the generated json looks like:
```
"securitySchemes" : {
     
      "OAuth2" : {
        "description" : "This API uses OAuth 2 with the implicit grant flow.",
        "flows" : [ {
          "authorizationUrl" : "https://api.example.com/oauth2/authorize",
          "scopes" : {
            "write_pets" : "modify pets in your account",
            "read_pets" : "read your pets"
          }
        }
    }
  }
```
according to the documentation https://swagger.io/docs/specification/authentication/oauth2/ it has to be like this
```
  "securityDefinitions": {
    "api_key": {
      "type": "apiKey",
      "name": "api_key",
      "in": "header"
    },
    "petstore_auth": {
      "type": "oauth2",
      "authorizationUrl": "https://petstore.swagger.io/oauth/authorize",
      "flow": "implicit",
      "scopes": {
        "read:pets": "read your pets",
        "write:pets": "modify pets in your account"
      }
    }
  },
```
this json is stolen from https://petstore.swagger.io/

without adding the flow type the authorization window will look like this
![Bildschirmfoto vom 2022-10-14 15-51-16](https://user-images.githubusercontent.com/40441078/195863629-f0f0bae9-b883-47ec-bd3c-536dc788d8bd.png)
but it should be like this
![Bildschirmfoto vom 2022-10-14 15-58-12](https://user-images.githubusercontent.com/40441078/195865143-e0d0595c-1d43-41e2-b0d8-7105bffcffe1.png)
as you can see instead of (OAuth,0) and Flow: 0 it is written (OAuth2, flowtype) and Flow: implicit the input for client_id is also missing.
Because of this all OAuth flows (authorizationCode, implicit, password or clientCredentials) will not work from the swaggerui.

Kind regards.

**Comment by dzikoysk** [comment]
Oh I see, the `"type": "oauth2"` is missing, thanks for reporting this!

**Comment by VladDrakul1986** [comment]
> Oh I see, the `"type": "oauth2"` is missing, thanks for reporting this! I'll fix it later today/tomorrow.

Sorry @dzikoysk  i shorten the json to much, the type is included.
what is missing is `"flow": "implicit"` and so on right now the json just has this:
```
"flows" : [ {
          "authorizationUrl"
```
iam not sure how OpenApi is evaluating this the documentation differs a bit from the petshop example. Find below the listing from the documentation.
```
# Step 1 - define the security scheme
components:
  securitySchemes:
    oAuthSample:    # <---- arbitrary name
      type: oauth2
      description: This API uses OAuth 2 with the implicit grant flow. [More info](https://api.example.com/docs/auth)
      flows:
        implicit:   # <---- OAuth flow(authorizationCode, implicit, password or clientCredentials)
          authorizationUrl: https://api.example.com/oauth2/authorize
          scopes:
            read_pets: read your pets
            write_pets: modify pets in your account
# Step 2 - apply security globally...
security: 
  - oAuthSample: 
    - write_pets
    - read_pets
# ... or to individual operations
paths:
  /pets:
    patch:
      summary: Add a new pet
      security: 
        - oAuthSample: 
          - write_pets
          - read_pets
```
the information of the line
`implicit:   # <---- OAuth flow(authorizationCode, implicit, password or c

**Comment by dzikoysk** [comment]
Thanks for explanation. Looks like the first scheme you've linked is written using OpenApi Specification v2:

```
  "securityDefinitions": {
    "api_key": {
      "type": "apiKey",
      "name": "api_key",
      "in": "header"
    },
    "petstore_auth": {
      "type": "oauth2",
      "authorizationUrl": "https://petstore.swagger.io/oauth/authorize",
      "flow": "implicit",
      "scopes": {
        "read:pets": "read your pets",
        "write:pets": "modify pets in your account"
      }
    }
  },
```

Looks like we should change `flows` array to object for v3.

**Comment by VladDrakul1986** [comment]
yes you are right the petshop example was still a v2 the v3 one https://petstore3.swagger.io/ has it like the official documentation :see_no_evil:

**Comment by dzikoysk** [comment]
That's how it looks now:

```json
"OAuth2" : {
  "description" : "This API uses OAuth 2 with the implicit grant flow.",
  "flows" : {
    "implicit" : {
      "authorizationUrl" : "https://api.example.com/oauth2/authorize",
      "scopes" : {
        "read_pets" : "read your pets",
        "write_pets" : "modify pets in your account"
      }
    }
  },
  "type" : "oauth2"
```

Fix is already available in 5.0.2-SNAPSHOT :)

**Comment by VladDrakul1986** [comment]
yes you are right the petshop example was still a v2 the v3 one https://petstore3.swagger.io/ has it like the official documentation :see_no_evil:
